### DISCUSSION TITLE
Recommanded max amount of files for a single property?

### DISCUSSION THREAD

**Discussion by NomadicDeveloper22** [pocketbase/pocketbase#3283] [Q&A] [answered]
I have an idea for a project still in the planning phase. I want to use pocketbase as sort of a middle man for accessing S3. Specifically using the authentication and for accessing files.

prototype collection structure (using 0.15.3 on pockethost.io for quick idea visualization)
![image](https://github.com/pocketbase/pocketbase/assets/32081703/971a3c1c-c474-4744-a7de-e152aa87e331)

But I acknowledge pocketbase isn't exactly designed for this use case. But is there any problem if say the user want's to add hundreds of files on that single "files" property? Or at least known issue?

**Comment by ganigeorgiev** [comment, marked as answer]
I don't think there will be any noticeable drawback whether a `file` field will have 100 or 1000 files as long as you have the db and storage space.

I'll be more concerned about the max upload file size. I haven't tested PocketBase with big files (eg. 1GB+) and I'm not really sure how well it will perform in practice, but you should be aware of the following limitations:
- Currently we have a default request read timeout of 10min. It can be changed programmatically when attaching to the `OnBeforeServe` hook and changing `e.Server.ReadTimeout`.
- There is no builtin support for partial uploads, aka. uploads where the user can pause/resume.
- There is no builtin cache and every time a file is requested via the PocketBase API we'll perform S3 API request to fetch it and stream it back to the client.
  This could trigger the rate limits for some S3 providers.
  A workaround could be to have a file cache on reverse proxy level OR serve the file directly with the S3 CDN (this will work only with public files; see also https://github.com/pocketbase/pocketbase/issues/28).
- This is not PocketBase specific, but serving a lot of big files can easily lead to additional server costs for excess outgoing network traffic.
